core/server/services/nft-oembed.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @typedef {import('./oembed').ICustomProvider} ICustomProvider
+ * @typedef {import('./oembed').IExternalRequest} IExternalRequest
+ */
+
+const OPENSEA_PATH_REGEX = /^\/assets\/(0x[a-f0-9]+)\/(\d+)/;
+
+/**
+ * @implements ICustomProvider
+ */
+class NFTOEmbedProvider {
+    /**
+     * @param {object} dependencies
+     */
+    constructor(dependencies) {
+        this.dependencies = dependencies;
+    }
+
+    /**
+     * @param {URL} url
+     * @returns {Promise<boolean>}
+     */
+    async canSupportRequest(url) {
+        return url.host === 'opensea.io' && OPENSEA_PATH_REGEX.test(url.pathname);
+    }
+
+    /**
+     * @param {URL} url
+     * @param {IExternalRequest} externalRequest
+     *
+     * @returns {Promise<import('oembed-parser').RichTypeData>}
+     */
+    async getOEmbedData(url, externalRequest) {
+        const [match, transaction, asset] = url.pathname.match(OPENSEA_PATH_REGEX);
+        if (!match) {
+            return null;
+        }
+        const result = await externalRequest(`https://api.opensea.io/api/v1/asset/${transaction}/${asset}/`, {
+            json: true
+        });
+        return {
```

Comment:
I'm trying to stick to the OEmbed spec here - but I'm not sure if that's how you imagined it?

What's particularly tricky with the OEmbed spec is that we should provide the minimum width & height in pixels - which I've hardcoded to 1000 atm :grimacing:

Reply:
Yeah, we should stick the the oembed spec as much as possible. We don't actually use the width/height for anything at the moment other than to verify that we have what looks like a [valid oembed response here](https://github.com/TryGhost/Ghost/blob/main/core/server/services/oembed.js#L255-L257) so hardcoded value is fine for now 👍